mirgecom/transport.py

```diff
@@ -0,0 +1,212 @@
+    r"""Transport model with uniform, constant properties.
+
+    .. automethod:: __init__
+
+    Inherits from (and implements) :class:`TransportModel`.
+    """
+
+    def __init__(self, bulk_viscosity=0, viscosity=0,
+                 thermal_conductivity=0,
+                 species_diffusivity=None):
+        """Initialize uniform, constant transport properties."""
+        if species_diffusivity is None:
+            species_diffusivity = np.empty((0,), dtype=object)
+        self._mu_bulk = bulk_viscosity
+        self._mu = viscosity
+        self._kappa = thermal_conductivity
+        self._d_alpha = species_diffusivity
+
+    def bulk_viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the bulk viscosity for the gas, $\mu_{B}$."""
+        return self._mu_bulk
+
+    def viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas dynamic viscosity, $\mu$."""
+        return self._mu
+
+    def viscosity2(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the 2nd viscosity coefficent, $\lambda$."""
+        return self._mu_bulk - 2 * self._mu / 3
+
+    def thermal_conductivity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas thermal_conductivity, $\kappa$."""
+        return self._kappa
+
+    def species_diffusivity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the vector of species diffusivities, ${d}_{\alpha}$."""
+        nspecies = len(cv.species_mass)
+        assert nspecies == len(self._d_alpha)
+        return self._d_alpha
+
+
+class PowerLawTransport(TransportModel):
+    r"""Transport model with simple power law properties.
+
+    Inherits from (and implements) :class:`TransportModel` based on a
+    temperature-dependent power law.
+
+    .. automethod:: __init__
+    .. automethod:: bulk_viscosity
+    .. automethod:: viscosity
+    .. automethod:: viscosity2
+    .. automethod:: species_diffusivity
+    .. automethod:: thermal_conductivity
+    """
+
+    # air-like defaults here
+    def __init__(self, alpha=0.6, beta=4.093e-7, sigma=2.5, n=.666,
+                 species_diffusivity=None):
+        """Initialize power law coefficients and parameters."""
+        if species_diffusivity is None:
+            species_diffusivity = np.empty((0,), dtype=object)
+        self._alpha = alpha
+        self._beta = beta
+        self._sigma = sigma
+        self._n = n
+        self._d_alpha = species_diffusivity
+
+    def bulk_viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the bulk viscosity for the gas, $\mu_{B}$.
+
+        $\mu_{B} = \alpha\mu$
+        """
+        return self._alpha * self.viscosity(eos, cv)
+
+    # TODO: Should this be memoized? Avoid multiple calls?
+    def viscosity(self, eos: GasEOS, cv: ConservedVars):
+        r"""Get the gas dynamic viscosity, $\mu$.
+
+        $\mu = \beta{T}^n$
+        """
+        actx = cv.array_context
+        gas_t = eos.temperature(cv)
+        return self._beta * actx.np.power(gas_t, self._n)
```

Comment:
Correct - but we (will) need this transport model.  In the follow-on #466, where transport is used in tests I have flagged it as non-working.  Not sure exactly how best to handle it.

Reply:
Would it make sense to leave this model out for now, and add it in a separate PR when it's fully working?

Otherwise maybe just `raise NotImplementedError` and add a comment explaining.

Reply:
If you can tolerate it, I would prefer to leave the implementation in, and just raise an exception if someone tries to instantiate one. My original plan was to just whip out an implementation for `power` in the array context, but that turned out to be a bigger bite than I could take right now.